### ISSUE TITLE
linter: vitest/require-mock-type-parameters false positive on typed vi.fn() with chained mockReturnValueOnce()

### ISSUE THREAD

**Issue by hratio** [oxc-project/oxc#22482]
### What version of Oxlint are you using?

`1.62.0`

### What command did you run?

pnpm oxlint --type-aware path/to/repro.test.ts -f json

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```
{
    "plugins": ["vitest"],
    "rules": {
      "vitest/require-mock-type-parameters": "warn"
    }
  }
```

### What happened?

vitest/require-mock-type-parameters reports a warning for a vi.fn call that already has an explicit type parameter when the mock is configured through a fluent .mockReturnValueOnce(...) chain.

**Minimal repro:**
```ts
import { vi } from "vitest";

type ResultAsync<T, E> = {
  readonly __ok?: T;
  readonly __err?: E;
};

declare const errAsync: <T, E>(error: E) => ResultAsync<T, E>;
declare const okAsync: <T, E>(value: T) => ResultAsync<T, E>;

const operation = vi
  .fn<() => ResultAsync<string, string>>()
  .mockReturnValueOnce(errAsync("error1"))
  .mockReturnValueOnce(errAsync("error2"))
  .mockReturnValueOnce(okAsync("success"));
```

**Actual diagnostic:**

vitest/require-mock-type-parameters: Missing type parameters on mock function call

The diagnostic span covers the whole chained expression starting at vi, even though the vi.fn call itself is typed:

```ts
.fn<() => ResultAsync<string, string>>()
```

**Expected result:**

No warning. The rule should only require type parameters on vi.fn() itself. In this example vi.fn already has a type parameter, and Vitest’s mockReturnValueOnce returns this, so TypeScript preserves the Mock<() => ResultAsync<string, string>> type through the chain.

**Additional context:**

Vitest’s type definitions make this type-safe:
```ts
declare function fn<T extends Procedure | Constructable = Procedure>(
  originalImplementation?: T
): Mock<T>;

mockReturnValueOnce(value: MockReturnType<T>): this;
```

**Comment by camc314** [comment]
Hi, thanks for reporting this issue.

It looks like this was already fixed on `main` by #22019, which changed `vitest/require-mock-type-parameters` to check type arguments on the specific `vi.fn`/import member call in a chain rather than the outer fluent call.

I checked your repro against `main` and it no longer reports.

upgrading to >1.63 should resolve the issue.

Please report any other issues (or if upgrading does not resolve the issue.

Thanks!

**Comment by hratio** [comment]
Thanks, I asked my LLM to double check issues beforehand and it said it couldn't find any. Sorry'bout that :-)
